lib/AVL-tree/AVL-tree.c

```diff
@@ -0,0 +1,308 @@
+#include "AVL-tree.h"
+#include "errorCodes.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+struct Node
+{
+    char* key;
+    char* value;
+    Node* rightChild;
+    Node* leftChild;
+    int height;
+};
+
+struct AVLTree
+{
+    Node* root;
+};
+
+AVLTree* createAVLTree()
+{
+    AVLTree* tree = calloc(1, sizeof(AVLTree));
+    if (tree == NULL)
+    {
+        return NULL;
+    }
+    tree->root = NULL;
+    return tree;
+}
+
+static unsigned int height(Node* node)
+{
+    return (node != NULL) ? node->height : 0;
+}
+
+static fixHeight(Node* node)
+{
+    unsigned int heightLeft = height(node->leftChild);
+    unsigned int heightRight = height(node->rightChild);
+    node->height = (heightLeft > heightRight ? heightLeft : heightRight) + 1;
+}
+
+static int balanceFactor(Node* node)
+{
+    return height(node->rightChild) - height(node->leftChild);
+}
+
+static Node* rotateRight(Node* node)
+{
+    Node* left = node->leftChild;
+    node->leftChild = left->rightChild;
+    left->rightChild = node;
+    
+    fixHeight(node);
+    fixHeight(left);
+
+    return left;
+}
+
+static Node* rotateLeft(Node* node)
+{
+    Node* right = node->rightChild;
+    node->rightChild = right->leftChild;
+    right->leftChild = node;
+
+    fixHeight(node);
+    fixHeight(right);
+
+    return right;
+}
+
+static Node* balance(Node** node)
+{
+    if ((*node) == NULL)
+    {
+        return NULL;
+    }
+
+    fixHeight((*node));
```

Comment:
```suggestion
    fixHeight(*node);
```
И ниже. Тут скобки не нужны